go test fails in pkg/log

```
mjog-macbookpro:log mjog$ go version
go version go1.9.2 darwin/amd64
mjog-macbookpro:log mjog$ go test -v .
=== RUN   TestBasic
=== RUN   TestBasic/0
=== RUN   TestBasic/1
=== RUN   TestBasic/2
=== RUN   TestBasic/3
=== RUN   TestBasic/4
=== RUN   TestBasic/5
=== RUN   TestBasic/6
=== RUN   TestBasic/7
=== RUN   TestBasic/8
=== RUN   TestBasic/9
=== RUN   TestBasic/10
=== RUN   TestBasic/11
=== RUN   TestBasic/12
=== RUN   TestBasic/13
=== RUN   TestBasic/14
=== RUN   TestBasic/15
=== RUN   TestBasic/16
=== RUN   TestBasic/17
=== RUN   TestBasic/18
=== RUN   TestBasic/19
=== RUN   TestBasic/20
=== RUN   TestBasic/21
2017-12-18 10:59:29.583602515 -0800 PST m=+0.013352556 write error: write /var/folders/j_/fq241wf558v_n2ph_dtmf06m00cbnf/T/log_test190309110: file already closed
--- FAIL: TestBasic (0.01s)
    --- FAIL: TestBasic/0 (0.00s)
    	log_test.go:100: Got '2017-12-18T10:59:29.571-0800	debug	Hello', expected a match with '.*Z	debug	Hello'
    --- FAIL: TestBasic/1 (0.00s)
    	log_test.go:100: Got '2017-12-18T10:59:29.572-0800	debug	Hello', expected a match with '.*Z	debug	Hello'
    --- FAIL: TestBasic/2 (0.00s)
    	log_test.go:100: Got '2017-12-18T10:59:29.573-0800	debug	Hello', expected a match with '.*Z	debug	Hello'
    --- FAIL: TestBasic/3 (0.00s)
    	log_test.go:100: Got '2017-12-18T10:59:29.573-0800	debug	Hello', expected a match with '.*Z	debug	Hello'
    --- FAIL: TestBasic/4 (0.00s)
    	log_test.go:100: Got '2017-12-18T10:59:29.573-0800	info	Hello', expected a match with '.*Z	info	Hello'
    --- FAIL: TestBasic/5 (0.00s)
    	log_test.go:100: Got '2017-12-18T10:59:29.574-0800	info	Hello', expected a match with '.*Z	info	Hello'
    --- FAIL: TestBasic/6 (0.00s)
    	log_test.go:100: Got '2017-12-18T10:59:29.575-0800	info	Hello', expected a match with '.*Z	info	Hello'
    --- FAIL: TestBasic/7 (0.00s)
    	log_test.go:100: Got '2017-12-18T10:59:29.576-0800	info	Hello', expected a match with '.*Z	info	Hello'
    --- FAIL: TestBasic/8 (0.00s)
    	log_test.go:100: Got '2017-12-18T10:59:29.576-0800	warn	Hello', expected a match with '.*Z	warn	Hello'
    --- FAIL: TestBasic/9 (0.00s)
    	log_test.go:100: Got '2017-12-18T10:59:29.577-0800	warn	Hello', expected a match with '.*Z	warn	Hello'
    --- FAIL: TestBasic/10 (0.00s)
    	log_test.go:100: Got '2017-12-18T10:59:29.577-0800	warn	Hello', expected a match with '.*Z	warn	Hello'
    --- FAIL: TestBasic/11 (0.00s)
    	log_test.go:100: Got '2017-12-18T10:59:29.578-0800	warn	Hello', expected a match with '.*Z	warn	Hello'
    --- FAIL: TestBasic/12 (0.00s)
    	log_test.go:100: Got '2017-12-18T10:59:29.578-0800	error	Hello', expected a match with '.*Z	error	Hello'
    --- FAIL: TestBasic/13 (0.00s)
    	log_test.go:100: Got '2017-12-18T10:59:29.579-0800	error	Hello', expected a match with '.*Z	error	Hello'
    --- FAIL: TestBasic/14 (0.00s)
    	log_test.go:100: Got '2017-12-18T10:59:29.579-0800	error	Hello', expected a match with '.*Z	error	Hello'
    --- FAIL: TestBasic/15 (0.00s)
    	log_test.go:100: Got '2017-12-18T10:59:29.580-0800	error	Hello', expected a match with '.*Z	error	Hello'
    --- FAIL: TestBasic/16 (0.00s)
    	log_test.go:100: Got '2017-12-18T10:59:29.580-0800	debug	Hello	{"key": "value"}', expected a match with '.*Z	debug	Hello	{"key": "value"}'
    --- FAIL: TestBasic/17 (0.00s)
    	log_test.go:100: Got '2017-12-18T10:59:29.581-0800	debug	log/log_test.go:65	Hello', expected a match with '.*Z	debug	log/log_test.go:.*	Hello'
    --- FAIL: TestBasic/18 (0.00s)
    	log_test.go:100: Got '{"level":"debug","time":"2017-12-18T10:59:29.581-0800","caller":"log/log_test.go:67","msg":"Hello","stack":"istio.io/istio/pkg/log.Debug\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log.go:152\nistio.io/istio/pkg/log.TestBasic.func19\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log_test.go:67\nistio.io/istio/pkg/log.TestBasic.func23.1\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log_test.go:91\nistio.io/istio/pkg/log.captureStdout\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log_test.go:218\nistio.io/istio/pkg/log.TestBasic.func23\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log_test.go:79\ntesting.tRunner\n\t/usr/local/Cellar/go/1.9.2/libexec/src/testing/testing.go:746"}', expected a match with '{"level":"debug","time":".*T.*Z","caller":"log/log_test.go:.*","msg":"Hello","stack":".*"}'
    --- FAIL: TestBasic/19 (0.00s)
    	log_test.go:100: Got '{"level":"info","time":"2017-12-18T10:59:29.582-0800","caller":"log/log_test.go:69","msg":"Hello","stack":"istio.io/istio/pkg/log.Info\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log.go:239\nistio.io/istio/pkg/log.TestBasic.func20\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log_test.go:69\nistio.io/istio/pkg/log.TestBasic.func23.1\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log_test.go:91\nistio.io/istio/pkg/log.captureStdout\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log_test.go:218\nistio.io/istio/pkg/log.TestBasic.func23\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log_test.go:79\ntesting.tRunner\n\t/usr/local/Cellar/go/1.9.2/libexec/src/testing/testing.go:746"}', expected a match with '{"level":"info","time":".*T.*Z","caller":"log/log_test.go:.*","msg":"Hello","stack":".*"}'
    --- FAIL: TestBasic/20 (0.00s)
    	log_test.go:100: Got '{"level":"warn","time":"2017-12-18T10:59:29.582-0800","caller":"log/log_test.go:71","msg":"Hello","stack":"istio.io/istio/pkg/log.Warn\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log.go:210\nistio.io/istio/pkg/log.TestBasic.func21\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log_test.go:71\nistio.io/istio/pkg/log.TestBasic.func23.1\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log_test.go:91\nistio.io/istio/pkg/log.captureStdout\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log_test.go:218\nistio.io/istio/pkg/log.TestBasic.func23\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log_test.go:79\ntesting.tRunner\n\t/usr/local/Cellar/go/1.9.2/libexec/src/testing/testing.go:746"}', expected a match with '{"level":"warn","time":".*T.*Z","caller":"log/log_test.go:.*","msg":"Hello","stack":".*"}'
    --- FAIL: TestBasic/21 (0.00s)
    	log_test.go:100: Got '{"level":"error","time":"2017-12-18T10:59:29.583-0800","caller":"log/log_test.go:73","msg":"Hello","stack":"istio.io/istio/pkg/log.Error\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log.go:181\nistio.io/istio/pkg/log.TestBasic.func22\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log_test.go:73\nistio.io/istio/pkg/log.TestBasic.func23.1\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log_test.go:91\nistio.io/istio/pkg/log.captureStdout\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log_test.go:218\nistio.io/istio/pkg/log.TestBasic.func23\n\t/Users/mjog/GOHOME/src/istio.io/istio/pkg/log/log_test.go:79\ntesting.tRunner\n\t/usr/local/Cellar/go/1.9.2/libexec/src/testing/testing.go:746"}', expected a match with '{"level":"error","time":".*T.*Z","caller":"log/log_test.go:.*","msg":"Hello","stack":".*"}'
=== RUN   TestEnabled
=== RUN   TestEnabled/0
=== RUN   TestEnabled/1
=== RUN   TestEnabled/2
=== RUN   TestEnabled/3
=== RUN   TestEnabled/4
--- PASS: TestEnabled (0.00s)
    --- PASS: TestEnabled/0 (0.00s)
    --- PASS: TestEnabled/1 (0.00s)
    --- PASS: TestEnabled/2 (0.00s)
    --- PASS: TestEnabled/3 (0.00s)
    --- PASS: TestEnabled/4 (0.00s)
=== RUN   TestOddballs
--- PASS: TestOddballs (0.00s)
=== RUN   TestCapture
=== RUN   TestCapture/0
=== RUN   TestCapture/1
=== RUN   TestCapture/2
--- FAIL: TestCapture (0.00s)
    --- FAIL: TestCapture/0 (0.00s)
    	log_test.go:202: Got '2017-12-18T10:59:29.585-0800	info	log/log_test.go:218	Hello', expecting to match '.*Z	info	log/log_test.go:.*	Hello'
    --- FAIL: TestCapture/1 (0.00s)
    	log_test.go:202: Got '2017-12-18T10:59:29.585-0800	info	log/log_test.go:186	There', expecting to match '.*Z	info	log/log_test.go:.*	There'
    --- FAIL: TestCapture/2 (0.00s)
    	log_test.go:202: Got '2017-12-18T10:59:29.585-0800	info	log/log_test.go:189	Goodbye', expecting to match '.*Z	info	log/log_test.go:.*	Goodbye'
=== RUN   TestOpts
=== RUN   TestOpts/0
=== RUN   TestOpts/1
=== RUN   TestOpts/2
=== RUN   TestOpts/3
=== RUN   TestOpts/4
=== RUN   TestOpts/5
=== RUN   TestOpts/6
=== RUN   TestOpts/7
=== RUN   TestOpts/8
=== RUN   TestOpts/9
=== RUN   TestOpts/10
=== RUN   TestOpts/11
=== RUN   TestOpts/12
--- PASS: TestOpts (0.00s)
    --- PASS: TestOpts/0 (0.00s)
    --- PASS: TestOpts/1 (0.00s)
    --- PASS: TestOpts/2 (0.00s)
    --- PASS: TestOpts/3 (0.00s)
    --- PASS: TestOpts/4 (0.00s)
    --- PASS: TestOpts/5 (0.00s)
    --- PASS: TestOpts/6 (0.00s)
    --- PASS: TestOpts/7 (0.00s)
    --- PASS: TestOpts/8 (0.00s)
    --- PASS: TestOpts/9 (0.00s)
    --- PASS: TestOpts/10 (0.00s)
    --- PASS: TestOpts/11 (0.00s)
    --- PASS: TestOpts/12 (0.00s)
=== RUN   TestLevel
=== RUN   TestLevel/0
=== RUN   TestLevel/1
=== RUN   TestLevel/2
=== RUN   TestLevel/3
=== RUN   TestLevel/4
--- PASS: TestLevel (0.00s)
    --- PASS: TestLevel/0 (0.00s)
    --- PASS: TestLevel/1 (0.00s)
    --- PASS: TestLevel/2 (0.00s)
    --- PASS: TestLevel/3 (0.00s)
    --- PASS: TestLevel/4 (0.00s)
FAIL
exit status 1
FAIL	istio.io/istio/pkg/log	0.049s
```

--- a/mixer/cmd/mixs/cmd/root.go
+++ b/mixer/cmd/mixs/cmd/root.go
@@ -43,6 +43,13 @@ func GetRootCmd(args []string, info map[string]template.Info, adapters []adapter
 	rootCmd.SetArgs(args)
 	rootCmd.PersistentFlags().AddGoFlagSet(flag.CommandLine)
 
+	// hack to make flag.Parsed return true such that glog is happy
+	// about the flags having been parsed
+	fs := flag.NewFlagSet("", flag.ContinueOnError)
+	/* #nosec */
+	_ = fs.Parse([]string{})
+	flag.CommandLine = fs
+
 	rootCmd.AddCommand(serverCmd(info, adapters, printf, fatalf))
 	rootCmd.AddCommand(crdCmd(info, adapters, printf, fatalf))
 	rootCmd.AddCommand(validatorCmd(info, adapters, printf, fatalf))

--- a/pkg/log/log.go
+++ b/pkg/log/log.go
@@ -52,6 +52,8 @@
 package log
 
 import (
+	"time"
+
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 	"go.uber.org/zap/zapgrpc"
@@ -112,8 +114,11 @@ func configure(options *Options, b builder) error {
 			LineEnding:     zapcore.DefaultLineEnding,
 			EncodeLevel:    zapcore.LowercaseLevelEncoder,
 			EncodeCaller:   zapcore.ShortCallerEncoder,
-			EncodeTime:     zapcore.ISO8601TimeEncoder,
 			EncodeDuration: zapcore.StringDurationEncoder,
+
+			EncodeTime: func(t time.Time, enc zapcore.PrimitiveArrayEncoder) {
+				enc.AppendString(t.UTC().Format("2006-01-02T15:04:05.000Z0700"))
+			},
 		},
 
 		OutputPaths:       options.OutputPaths,

